lib/train/file.rb

```diff
@@ -147,5 +129,85 @@ def mounted?
 
       !mounted.nil? && !mounted.stdout.nil? && !mounted.stdout.empty?
     end
+
+    def md5sum
+      return perform_ruby_checksum(method) if defined?(@use_ruby_checksum)
+
+      @md5_command ||= case @backend.os.family
+                       when 'darwin'
+                         # `-r` reverses output to match `md5sum` behavior
+                         'md5 -r'
+                       when 'solaris'
+                         'digest -a md5'
+                       else
+                         'md5sum'
+                       end
+
+      return perform_checksum(:md5) unless @backend.os.family == 'windows'
+
+      perform_checksum_windows(:md5)
+    end
+
+    def sha256sum
+      return perform_ruby_checksum(method) if defined?(@use_ruby_checksum)
+
+      @sha256_command ||= case @backend.os.family
+                          when 'darwin', 'hpux', 'qnx'
+                            'shasum -a 256'
+                          when 'solaris'
+                            'digest -a sha256'
+                          else
+                            'sha256sum'
+                          end
+
+      return perform_checksum(:sha256) unless @backend.os.family == 'windows'
+
+      perform_checksum_windows(:sha256)
+    end
+
+    private
+
+    def perform_checksum(method)
+      case method
+      when :md5
+        cmd = "#{@md5_command} #{@path}"
+      when :sha256
+        cmd = "#{@sha256_command} #{@path}"
+      end
+
+      res = @backend.run_command(cmd)
+      return res.stdout.split(' ').first if res.exit_status == 0
+
+      perform_ruby_checksum(method)
```

Comment:
I would return an error for that method instead of calling `perform_ruby_checksum` here